packages/design-system/src/components/tree/tree.tsx

```diff
@@ -439,11 +441,25 @@ const useKeyboardNavigation = <Data extends { id: string }>({
     }
   }, [root, rootRef, selectedItemSelector, setFocus]);
 
+  // When we stop editing an item, we want to focus it back again.
+  // To continue keyboard navigation.
+  useEffect(() => {
```

Comment:
After clicking Enter we save item somewhere, so we have an event to save, we can focus in same event handler, without using useEffect IMO

Reply:
Oh is it, let me try that way then 👍

Reply:
The other way is to return focus inside useContentEditable, preserving it initially

Reply:
Anyway this is just ideas how to not create new useEffect

Reply:
Yes, the change in `use-content-editable` might not be the best option. Since it focusses the text node. But not the whole tree-node. Anyways i will try the options for doing it without `useEffect` 👍